src/main/java/pl/vm/javaguild/designpatterns/pattern/creational/prototype/Prototype.java

```diff
@@ -0,0 +1,27 @@
+package pl.vm.javaguild.designpatterns.pattern.creational.prototype;
+
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.exception.CloneNotSupportedRuntimeException;
+
+/**
+ * Main definition of prototype design pattern.
+ * It provides cloning logic via {@link Prototype#clone()} ()}
+ *
+ * @param <T> generic type to be cloned
+ */
+public abstract class Prototype<T> implements Cloneable {
+
+    /**
+     * Create <b>shallow copy</b> of this object.
+     * <p>
+     * It should be overridden, whenever special logic should be added.
+     *
+     * @return copy of this
+     */
+    public T copy() {
+        try {
+            return (T) super.clone();
+        } catch (CloneNotSupportedException e) {
+            throw new CloneNotSupportedRuntimeException("Cannot copy current object: " + this);
```

Comment:
Tutaj mam problem z tym, ze CloneNotSupportedException e jest ignorowane, jak nie chcesz go podac dalej do CloneNotSupportedRuntimeException to przynajmniej bym zrobil log.error;

Tez konstruktor CloneNotSupportedRuntimeException jest dosc liberalny bo mozna wszystko tam wrzucic. Zastanowil bym sie czy nie przeniesc stalej String do CloneNotSupportedRuntimeException i podawac tylko Object jako argument. Lub jakies podobne rozwiazanie